projects/plugins/boost/app/modules/cache/Page_Cache.php

```diff
@@ -5,7 +5,18 @@
 use Automattic\Jetpack_Boost\Contracts\Is_Always_On;
 use Automattic\Jetpack_Boost\Contracts\Pluggable;
 
+require_once __DIR__ . '/Boost_Cache_Settings.php';
+
 class Page_Cache implements Pluggable, Is_Always_On {
+	/*
+	 * @var array - The settings for the page cache.
+	 */
+	private $settings = array();
+
+	public function __construct() {
+		$this->settings = new Boost_Cache_Settings();
+	}
```

Comment:
So, `Page_Cache` is used when visiting admin pages, and `Boost_Cache` is used when actually driving the caching process. Will they ever co-exist? Is it worth making Settings a singleton, or will only one exist at a time?

Reply:
I just realised that the code that defines REST_REQUEST runs after Boost_Cache has been created, so for those requests, the output buffer is created. It's not saved in the callback though, as it recognises it's a REST request.

The Boost_Cache constructor could check if the REQUEST_URI is /wp-admin/ or /wp-json/ but that only helps if the site is at "/". A page URL might have either of those strings in it.

So, the settings object is created twice in wp-admin and wp-json requests. I think it might be worth making the settings class a singleton.